### ISSUE TITLE
iOS allows too many characters

### ISSUE THREAD

**Issue by scottmas** [benhurott/react-native-masked-text#68]
The following custom definition permits too many characters on iOS. Android correctly forbids the user from entering more than the specified mask. In the following example, Android restricts the text input from having more than 1 digit, but iOS does not. 

```
 <TextInputMask
            type='custom'
            options={{mask: '9'}}
            value={this.state.inputVal}
            keyboardType='numeric'
            onChangeText={val => val.length <= 1 && this.setState({ inputVal: val })}
          />
```

**Comment by benhurott** [comment]
Hi,

After a long investigation, I realize it can be a bug in react-native.

I tried with both TextInputMask and default TextInput.
The state remains correct but it does not update current text input value O.o.

This is the code:

```jsx
export default class App extends React.Component {
  state = {
    inputVal: ''
  }

  _updateVal(val) {
    if (val.length <= 1) {
      this.setState({ inputVal: val }, () => {
        console.log(this.state)
      })
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>TextInputMask: </Text>

        <TextInputMask
          type='custom'
          options={{ mask: '9' }}
          value={this.state.inputVal}
          keyboardType='numeric'
          onChangeText={this._updateVal.bind(this)}
          style={styles.input}
        />

        <Text>Default TextInput: </Text>
        <TextInput
          value={this.state.inputVal}
          keyboardType='numeric'
          onChangeText={this._updateVal.bind(this)}
          style={styles.input}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  input: {
    borderWidth: 1,
    borderColor: 'red',
    height: 50,
    width: '90%'
  }
});
```

And take a look when we input some values into input:
![img_2647](https://user-images.githubusercontent.com/3187772/39099195-bd8559a0-464c-11e8-9df7-0321ef26d8c7.PNG)


Now, take a look at the `console.log` to see the state:
![image](https://user-images.githubusercontent.com/3187772/39099198-d9599b46-464c-11e8-8609-e35b7373748e.png)


This is absolutely bizarre and the same happen if we input values on MaskedTextInput.
I will open an issue on React-Native's repository.

Thanks for reporting.

**Comment by benhurott** [comment]
This issue is opened on React-Native's github:
[LINK To ISSUE](https://github.com/facebook/react-native/issues/18844)

**Comment by zaguini** [comment]
Any news?

**Comment by zaguini** [comment]
Hey, I've found a "workaround" for now: just define `maxLength` on your `TextInput`.

That's the "solution" while there isn't an official bugfix... Example:

If you define the `credit-card` mask, just define the `maxLength` of your input to `19`, so it will cover the 16 digits of the credit card plus the 3 spaces. :+1:

**Comment by benhurott** [comment]
Guys! Good news!
[This PR](https://github.com/facebook/react-native/pull/18278) solves this problem!

To solve this problem without painful merge, just merge this PR in the `ios` project

OR

1 - Open this file:
![image](https://user-images.githubusercontent.com/3187772/45257518-c0b26200-b37d-11e8-8c89-02b5ef70503d.png)

2 - Add this code:
![image](https://user-images.githubusercontent.com/3187772/45257527-dc1d6d00-b37d-11e8-82d7-13b99e2ed1e6.png)

3 - And this code:
![image](https://user-images.githubusercontent.com/3187772/45257538-0838ee00-b37e-11e8-977a-b15b77509114.png)


I will keep this issue opened until official fix.

**Comment by benhurott** [comment]
Fixed in [RN 0.57](https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md#ios-specific-fixes)
